index.js

```diff
@@ -51,6 +51,23 @@ exports.loginHandler = function (req, res, next) {
   }
 };
 
+exports.loginHandler = function (req, res, next) {
+  if (validator.isEmail(req.body.username)) {
+    User.find({ username: req.body.username, password: req.body.password }, function (err, users) {
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; NoSQL Injection

Unsanitized input from `the HTTP request body` flows into `find`, where it is used in an NoSQL query. This may result in an NoSQL Injection vulnerability.

<h6>
Line 56 | <a href="https://cwe.mitre.org/data/definitions/943.html">CWE-943</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 806</a> | <a href="https://learn.snyk.io/lesson/nosql-injection-attack/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 6 steps</b></summary>

`Step 1 - 3` https://github.com/snykyzach/zach-goof/blob/ae53e40227a57442d3bd9ad65a29be3112eda5f8/index.js#L55

`Step 4 - 6` https://github.com/snykyzach/zach-goof/blob/ae53e40227a57442d3bd9ad65a29be3112eda5f8/index.js#L56

</details>